tests/subsys/event_manager_proxy/src/data.c

```diff
@@ -55,25 +56,27 @@ static void test_data_response(void)
 	int err = k_sem_take(&waiting_response_sem, K_SECONDS(1));
 	zassert_ok(err, "No data response event received");
 
-	zassert_equal(event->val1, data_response.val1, "Unexpected value in response");
-	zassert_equal(event->val2, data_response.val2, "Unexpected value in response");
-	zassert_equal(event->val3, data_response.val3, "Unexpected value in response");
-	zassert_equal(event->val1u, data_response.val1u, "Unexpected value in response");
-	zassert_equal(event->val2u, data_response.val2u, "Unexpected value in response");
-	zassert_equal(event->val3u, data_response.val3u, "Unexpected value in response");
+	zassert_equal(rand_test_data.val1, data_response.val1, "Unexpected value in response");
+	zassert_equal(rand_test_data.val2, data_response.val2, "Unexpected value in response");
+	zassert_equal(rand_test_data.val3, data_response.val3, "Unexpected value in response");
+	zassert_equal(rand_test_data.val1u, data_response.val1u, "Unexpected value in response");
+	zassert_equal(rand_test_data.val2u, data_response.val2u, "Unexpected value in response");
+	zassert_equal(rand_test_data.val3u, data_response.val3u, "Unexpected value in response");
 
 	test_end(TEST_DATA_RESPONSE);
 }
 
 static void test_data_big_response(void)
 {
 	struct data_big_event *event = new_data_big_event();
+	struct data_big_content rand_test_data;
 
 	for (size_t n = 0; n < DATA_BIG_EVENT_BLOCK_SIZE; ++n) {
 		event->block[n] = sys_rand32_get();
 	}
 
 	memset(&data_big_response, 0, sizeof(data_big_response));
+	memcpy(&rand_test_data.block, &event->block, sizeof(data_big_response));
```

Comment:
Put sizeof of the actual target. This one is risky if anything changes.

Reply:
Should we also somehow ensure that both pointed data blocks have the same length (keep in mind these are separate types)?